### PR TITLE
wallet: add output info to change output in FundPsbt

### PR DESCRIPTION
**Pull request by guggero**
In case we create a change output when funding a PSBT we want to include its derivation path and public key information in the PSBT in order to make the output information more useful and verbose.

This is used in order to create exclusion proofs for our change addresses in Taro when funding any anchor TX through PSBT.

cc @bhandras.

### CODE REVIEW

File: wallet/psbt.go [hunk 1]
```diff
@@ -294,6 +309,52 @@ func addInputInfoSegWitV1(in *psbt.PInput, utxo *wire.TxOut,
 	}}
 }
 
+// addOutputInfo adds the BIP32 derivation info for an output from our internal
+// wallet.
+func addOutputInfo(out *psbt.POutput, txOut *wire.TxOut,
```
Comment (bhandras) [inline wallet/psbt.go:314]: nit: could be slightly more readable as ```getOutputInfo(*wire.TxOut, waddrmgr.ManagedPubKeyAddres) (*psbt.POutput, error)```

File: wallet/psbt_test.go [hunk 2]
```diff
@@ -293,6 +293,19 @@ func TestFundPsbt(t *testing.T) {
 				t, tc.expectedChangeBeforeFee,
 				txOuts[changeIndex].Value+int64(expectedFee),
 			)
+
+			changeTxOut := txOuts[changeIndex]
+			changeOutput := packet.Outputs[changeIndex]
+			require.NotEmpty(t, changeOutput.Bip32Derivation)
```
Comment (bhandras) [inline wallet/psbt_test.go:299]: nit: could also add some additional checks apart from `NotEmpty`.

**Review by bhandras — APPROVED** [review]
LGTM :tada:

File: wallet/psbt.go [hunk 3]
```diff
@@ -294,6 +309,52 @@ func addInputInfoSegWitV1(in *psbt.PInput, utxo *wire.TxOut,
...
+		},
+	}
+	out.Bip32Derivation = []*psbt.Bip32Derivation{
+		derivation,
+	}
+
+	// Include the Taproot derivation path as well if this is a P2TR output.
+	if txscript.IsPayToTaproot(txOut.PkScript) {
+		out.TaprootBip32Derivation = []*psbt.TaprootBip32Derivation{{
+			XOnlyPubKey:          derivation.PubKey[1:],
```
Comment (Roasbeef) [inline wallet/psbt.go:348]: Style nit: can capture this as `schnorrKey` above since it's used below also for `TaprootInternalKey`.

**Review by Roasbeef — COMMENTED** [review]
Easy diff, reads well, just a small comment in addition to what Andras left.

File: wallet/psbt.go (hunk 1 above)
Comment (guggero) [reply wallet/psbt.go]: Yeah, makes sense.

**Review by Roasbeef — APPROVED** [review]
LGTM ⏰

**Pull request merged**